apple/internal/binary_support.bzl

```diff
@@ -193,141 +181,7 @@ def _add_exported_symbols_lists(name, exported_symbols_lists_value):
 
     return exported_symbols_list_deps
 
-def _create_binary(
-        name,
-        platform_type,
-        product_type,
-        binary_type = "executable",
-        bundle_loader = None,
-        extension_safe = False,
-        link_swift_statically = False,
-        sdk_frameworks = [],
-        suppress_entitlements = False,
-        target_name_template = "%s.__internal__.apple_binary",
-        **kwargs):
-    """Creates a binary target for a bundle by linking user code.
-
-    This function also wraps the entitlements handling logic. It returns a
-    modified copy of the given keyword arguments that has `binary` and
-    `entitlements` attributes added if necessary and removes other
-    binary-specific options (such as `linkopts`).
-
-    Args:
-      name: The name of the bundle target, from which the binary target's name
-          will be derived.
-      platform_type: The platform type for which the binary should be built.
-      product_type: The product type of the bundle for which the binary will be built.
-      binary_type: The type of binary to create. Can be "executable",
-          "loadable_bundle" or "dylib".
-      bundle_loader: Label to an apple_binary target that will act as the
-          bundle_loader for this apple_binary. Can only be set if binary_type is
-          "loadable_bundle".
-      extension_safe: If true, compiles and links this framework with
-          '-application-extension', restricting the binary to use only
-          extension-safe APIs. False by default.
-      link_swift_statically: True/False, indicates whether the static versions of
-          the Swift standard libraries should be used during linking.
-      sdk_frameworks: Additional SDK frameworks that should be linked with the
-          final binary.
-      suppress_entitlements: True/False, indicates that the entitlements() should
-          be suppressed.
-      target_name_template: A string that will be used to derive the name of the
-          `apple_binary` target. This string must contain a single `%s`, which
-          will be replaced by the name of the bundle target.
-      **kwargs: The arguments that were passed into the top-level macro.
-
-    Returns:
-      A modified copy of `**kwargs` that should be passed to the bundling rule.
-    """
-    bundling_args = dict(kwargs)
-
-    rule_descriptor = rule_support.rule_descriptor_no_ctx(platform_type, product_type)
-
-    linkopts = kwargs.pop("linkopts", [])
-    linkopts = linkopts + collections.before_each("-rpath", rule_descriptor.rpaths)
-    linkopts = linkopts + rule_descriptor.extra_linkopts
-    if extension_safe:
-        linkopts = linkopts + ["-application_extension"]
```

Comment:
Note this change is wiped out, we probably need to reapply it elsewhere

Reply:
or is it OK because we pass `-fapplication-extension` when needed?

Reply:
I think for that to matter we also have to have the linkopt

Reply:
https://github.com/bazelbuild/rules_apple/pull/925/files#diff-eebcb98abfda3b6dca6c52be04dd76b3R271 it's there?

Reply:
that's only for extensions, this also applied to  ios_framework targets, pushed a commit that should restore this

Reply:
nice, though ideally we'd have a test 🙃

Reply:
it doesn't seem like there's a way to use starlark tests to validate flags atm, is that what you've seen too?